lib/ui/common/headerAndFooterListView.dart

```diff
@@ -6,56 +6,86 @@ Simple Widget to add header and footer elements to a list of elements
 It should ensure elements are built as needed and not the whole list.
 It hides the data offsetting complexity 
 */
-class HeaderAndFooterListView extends StatelessWidget implements ListViewSection {
-
+class HeaderAndFooterListView extends StatelessWidget
+    implements ListViewSection {
   final IndexedWidgetBuilder _containedItemBuilder;
   final int _containedItemCount;
   final List<Widget> _headers;
   final List<Widget> _footers;
   final ScrollPhysics _physics;
   final bool _shrinkWrap;
+  final Widget _contentDivider;
 
-  HeaderAndFooterListView({IndexedWidgetBuilder itemBuilder, int itemCount = 0, List<Widget> headers = const [], List<Widget>  footers = const [], ScrollPhysics physics, bool shrinkWrap}) : this._containedItemBuilder = itemBuilder, 
-  this._containedItemCount = itemCount, this._headers = headers, this._footers = footers, this._physics = physics, this._shrinkWrap = shrinkWrap;
+  HeaderAndFooterListView({
+    IndexedWidgetBuilder itemBuilder,
+    int itemCount = 0,
+    List<Widget> headers = const [],
+    List<Widget> footers = const [],
+    ScrollPhysics physics,
+    bool shrinkWrap,
+    Widget contentDivider,
+  })  : this._containedItemBuilder = itemBuilder,
+        this._containedItemCount = itemCount,
+        this._headers = headers,
+        this._footers = footers,
+        this._physics = physics,
+        this._shrinkWrap = shrinkWrap,
+        this._contentDivider = contentDivider;
 
   /*
   Modeling with lists for headers and footers simplifys the logic here and may be useful in the future
   For now we only expose the single header and footer functionality
   */
 
-    @override
+  @override
   Widget build(BuildContext context) {
-    return ListView.builder(
-      physics: _physics,
-      shrinkWrap: _shrinkWrap,
-      itemCount: itemCount(),
-      itemBuilder: itemBuilder(),
-    );
+    if (_contentDivider != null) {
+      return ListView.separated(
+        separatorBuilder: (context, index) {
+          if ((_headers.length > 0 && index < _headers.length) ||
+              (_footers.length > 0 &&
+                  index >= (_containedItemCount + _headers.length) - 1)) {
```

Comment:
Can we move this logic to a method? Have something like, `bool hasDivider()` that returns if the Divider should be shown or not.

Reply:
Done